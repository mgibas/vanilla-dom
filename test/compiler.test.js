const compiler = require('../compiler')
let attributesCompiler = require('../attributes-compiler')
attributesCompiler.compile = jest.fn()
attributesCompiler.compile.mockReturnValue('COMPILED-ATTRIBUTES')

describe('compiler', () => {
  describe('on compile', () => {
    it('empty source', () => {
      expect(compiler.compile('')).toMatchSnapshot();
    })
    it('single tag node', () => {
      expect(compiler.compile('<div></div>')).toMatchSnapshot();
    })
    it('single tag node with attributes', () => {
      expect(compiler.compile('<div foo="5" bar="some"></div>')).toMatchSnapshot();
    })
    it('multiple tag nodes', () => {
      expect(compiler.compile('<div></div><p></p>')).toMatchSnapshot();
    })
    it('text nodes', () => {
      expect(compiler.compile(`
        <div></div>
        some text

        ^ and empty line

        <p></p>
      `)).toMatchSnapshot();
    })
    it('nested nodes', () => {
      expect(compiler.compile(`
        <div>
          <p>hello</p>
        </div>
      `)).toMatchSnapshot();
    })
  })
})
