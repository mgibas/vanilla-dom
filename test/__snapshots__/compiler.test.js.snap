// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler on compile empty source 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

          
    }"
`;

exports[`compiler on compile multiple tag nodes 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

      
      (() => {
        let domRoot_el = document.createElement('div');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
    
      (() => {
        let domRoot_el = document.createElement('p');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
        
    }"
`;

exports[`compiler on compile nested nodes 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

      
      (() => {
        let tel = document.createTextNode(\`
        \`)
        domRoot.appendChild(tel)
      })();
    
      (() => {
        let domRoot_el = document.createElement('div');
        COMPILED-ATTRIBUTES
        
      (() => {
        let tel = document.createTextNode(\`
          \`)
        domRoot_el.appendChild(tel)
      })();
    
      (() => {
        let domRoot_el_el = document.createElement('p');
        COMPILED-ATTRIBUTES
        
      (() => {
        let tel = document.createTextNode(\`hello\`)
        domRoot_el_el.appendChild(tel)
      })();
        

        domRoot_el.appendChild(domRoot_el_el)
      })();
    
      (() => {
        let tel = document.createTextNode(\`
        \`)
        domRoot_el.appendChild(tel)
      })();
        

        domRoot.appendChild(domRoot_el)
      })();
    
      (() => {
        let tel = document.createTextNode(\`
      \`)
        domRoot.appendChild(tel)
      })();
        
    }"
`;

exports[`compiler on compile single tag node 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

      
      (() => {
        let domRoot_el = document.createElement('div');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
        
    }"
`;

exports[`compiler on compile single tag node with attributes 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

      
      (() => {
        let domRoot_el = document.createElement('div');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
        
    }"
`;

exports[`compiler on compile text nodes 1`] = `
"

    export mount = (domRoot, state) => {
      const reactivePaths = []

      
      (() => {
        let tel = document.createTextNode(\`
        \`)
        domRoot.appendChild(tel)
      })();
    
      (() => {
        let domRoot_el = document.createElement('div');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
    
      (() => {
        let tel = document.createTextNode(\`
        some text

        ^ and empty line

        \`)
        domRoot.appendChild(tel)
      })();
    
      (() => {
        let domRoot_el = document.createElement('p');
        COMPILED-ATTRIBUTES
            

        domRoot.appendChild(domRoot_el)
      })();
    
      (() => {
        let tel = document.createTextNode(\`
      \`)
        domRoot.appendChild(tel)
      })();
        
    }"
`;
